core/src/validator.rs

```diff
@@ -1719,115 +1719,111 @@ impl Validator {
     }
 
     pub fn join(self) {
-        drop(self.bank_forks);
         drop(self.cluster_info);
 
-        self.poh_service.join().expect("poh_service");
+        macro_rules! join_then_log {
```

Comment:
This seems like a separate change to assist in debugging thread-joining issues. Probably a good change to make, but not related to fixing this issue afiact?

Reply:
yeah. another piggy-bucked salvation from the ancient pr #593 by procrastinating me.